.github/scripts/make_mocs.py

```diff
@@ -0,0 +1,233 @@
+
+
+class MocMaker:
+    """
+    Class that generates the content of a MOC file
+    
+    That is, it generates a Markdown list of links to files and directories
+    """
+    def make_moc_for_files(self, directory, files):
+        output = ''
+        moc_filter = MocFileAndDirectoryFilter()
+        for file in files:
+            if not moc_filter.include_file_in_moc(directory, file):
+                continue
+            output += self.make_line_for_file(directory, file)
+        return output
+
+    def make_moc_for_sub_directories(self, directory, sub_directories):
+        output = ''
+        moc_filter = MocFileAndDirectoryFilter()
+        for sub_directory in sub_directories:
+            if not moc_filter.include_directory_in_moc(sub_directory):
+                continue
+            output += self.make_line_for_sub_directory(directory, sub_directory)
+        return output
+
+    def make_moc_for_directory(self, root, dirs, files):
+        result = ''
+
+        sorted_dirs = sorted(dirs, key=str.casefold)
+        result += self.make_moc_for_sub_directories(root, sorted_dirs)
+
+        sorted_files = sorted(files, key=str.casefold)
+        result += self.make_moc_for_files(root, sorted_files)
+
+        if not result:
+            result = '\n'
+        return result
+
+    def make_moc_for_directory_with_delimiters(self, root, dirs, files):
+        result = ''
+        result += MocDelimiter().initial_delimiter()
+        result += self.make_moc_for_directory(root, dirs, files)
+        result += MocDelimiter().final_delimiter()
+        return result
+
+    def update_existing_moc(self, initial_content, new_moc_content_with_delimiters):
+        inside_old_index = False
+        index_written = False
+        result = ''
+        for line in initial_content:
+
+            if line == MocDelimiter().initial_delimiter():
+                inside_old_index = True
+                result += new_moc_content_with_delimiters
+                index_written = True
+                continue
+
+            if line == MocDelimiter().final_delimiter():
+                inside_old_index = False
+                continue
+
+            if not inside_old_index:
+                result += line
+        if not index_written:
+            result += new_moc_content_with_delimiters
+
+        return result
+
+    def make_line_for_file(self, directory, file):
+        link_name, extension = os.path.splitext(file)
+        if extension != '.md':
+            link_name += extension
+        return self.make_link_line(directory, link_name)
+
+    def make_line_for_sub_directory(self, directory, sub_directory):
+        path = directory + '/' + sub_directory
+        namer = MocFileNamer()
+        file = namer.moc_name_for_sub_directory(sub_directory)
+        return self.make_link_line(path, file)
+
+    def strip_parent_directories_from_directory(self, directory):
+        # Ugly hack because all directory names start with '../../'
```

Comment:
you could use `os.path` perhaps?

Reply:
That comment is terribly unclear, and I need to remove the limitation that the script only works when run 2 directories down from the project root.

I should have added a comment saying that I'm using forward slashes throughout, as even if run on Windows, we would want unix-style `/` used in the generated Markdown.

Do you think it's a priority to test it on Windows, or is it mainly likely to be run on Unix, for the immediately foreseeable future?

(I can later add running the tests on Windows CI, if you feel it's worth it)

Reply:
I've added a couple of comments saying that I'm using forward slashes intentionally.